app/controllers/laboratories_controller.rb

```diff
@@ -23,13 +37,64 @@ def create
   def update
     redirect_to(root_url) && return unless current_user.can_edit_patient_laboratories?
 
+    patient_id = params.permit(:patient_id)[:patient_id]&.to_i
+
+    # Check if Patient ID is valid
+    unless Patient.exists?(patient_id)
+      error_message = "Lab Result cannot be updated for unknown monitoree with ID: #{patient_id}"
+      render(json: { error: error_message }, status: :bad_request) && return
+    end
+
+    # Check if user has access to patient
+    unless current_user.get_patient(patient_id)
+      error_message = "User does not have access to Patient with ID: #{patient_id}"
+      render(json: { error: error_message }, status: :forbidden) && return
+    end
+
     lab = Laboratory.find_by(id: params.permit(:id)[:id])
     lab.update!(lab_type: params.permit(:lab_type)[:lab_type],
                 specimen_collection: params.permit(:specimen_collection)[:specimen_collection],
                 report: params.permit(:report)[:report],
                 result: params.permit(:result)[:result])
-    History.lab_result_edit(patient: params.permit(:patient_id)[:patient_id],
+
+    History.lab_result_edit(patient: patient_id,
                             created_by: current_user.email,
                             comment: "User edited a lab result (ID: #{lab.id}).")
   end
+
+  # Delete an existing lab result
+  def destroy
+    redirect_to(root_url) && return unless current_user.can_edit_patient_laboratories?
```

Comment:
can we add all these actions as `before_action` to clean up the controller - I did this in the histories controller

Reply:
Good catch, I meant to do that after I finished writing tests and forgot.